File: web/src/models/FinancialRecord.ts
export enum RecordType{
    ShortTermAsset,
    LongTermAsset,
    ShortTermLiability,
    LongTermLiability
}

export default class FinancialRecord{
    id: number | undefined;
    value: number | undefined;
    // currency: string | undefined;
    name: string | undefined;
    rate: number | undefined;
    monthlyPayment: number | undefined;
    recordType: RecordType | undefined;
}

File: api/DemoApi/Models/FinancialRecord.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DemoApi.Models
{
    public class FinancialRecord
    {
        public int Id { get; set; }
        public double Value { get; set; }
        public string Currency { get; set; }
        public string Name { get; set; }
        public double Rate { get; set; }
        public double MonthlyPayment { get; set; }
        public RecordType RecordType { get; set; }
    }
}
